packages/opentelemetry-types/src/trace/trace_options.ts

```diff
@@ -14,6 +14,10 @@
  * limitations under the License.
  */
 
-export interface TraceOptions {
-  // TODO
-}
+/** Bit to represent whether trace is sampled in trace options. */
+export const TRACE_OPTIONS_SAMPLED = 0x1;
```

Comment:
Last minute nitpick, but since the file exports the `TraceOptions`, it feels redundant to specify it again here. Can we change it to just `SAMPLED` or `UNSAMPLED`?

Reply:
In same package it makes sense, how about when we import in other packages like core or instrumentations.

Reply:
Isn't it still `TraceOptions.SAMPLED` then?

Reply:
If I understand correctly, you are saying that the constants will be re-exported as-is. I don't think that's good practice from a separation of concern perspective. I would be inclined to do like @danielkhan mentioned, or if we want in some cases to export the constants for something private, then it could be in some other file that contains only the public constants that are not defined on public interfaces. However, I think it's achievable that there is always an interface exported to hold the constants.

Reply:
What about something like this ([typescript playground link](https://www.typescriptlang.org/play/#code/C4TwDgpgBAKgTgQwMYQPJmASwPYDsDOUAvFAAwAepAPhQIwDcAUBLgK4C2UAygu2ADaZcAc3RY8hAN6MosqAFUAclwCCAWQAKAGQCiAEWJlKAGhlzVm3QZJ1TAX0ZIJwKOQBcsRCjE4ChnnyCIj4SAHQW2vr0QA))?

```
export type TraceOptions = 0x0|0x1;
export enum SamplingOptions {
    UNSAMPLED = 0x0,
    SAMPLED = 0x1,
}
```

Note that if we just call them `SAMPLED` and `UNSAMPLED` then their namespacing will be lost when they are re-exported for the package as a whole. I.e. you would import it with `import {SAMPLED} from '@opentelemetry/types';`

Reply:
>  if we just call them SAMPLED and UNSAMPLED then their namespacing will be lost when they are re-exported for the package as a whole

This is what I was trying to highlight. Thanks @draffensperger for detailed explanation. I like the idea of `SampleOptions` (or some other name) enum.

Reply:
Is there a way to merge both in a single definition? Currently they basically represent the same concept but with different structures. Also, it would be easier to use if all possible `TraceOptions` flags are in the same enum since it will make discovery of possible values simpler.

Reply:
Good thought! @mayurkale22 what if we just call the enum `TraceOptions` then and remove the union type?

If someday there are more complicated combinations of bits that get specified we can come back and refactor, but I think there is plenty else to specify and work on in the mean time!